src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsNavigationSource.cs

```diff
@@ -87,6 +90,20 @@ public IEnumerable<IEdmNavigationPropertyBinding> NavigationPropertyBindings
             get { return this.navigationTargetsCache.GetValue(this, ComputeNavigationTargetsFunc, null); }
         }
 
+        public IEdmEntityType EntityType
+        {
+            get
+            {
+                if (!this.entityTypeSet)

```

Comment:
Why not check for !this.entityType  == null ?

Reply:
@wachugamaina Because the value can be null, I didn't want the expression `this.Type?.AsElementType() as IEdmEntityType;` to be executed multiple times since this property is referenced a lot across the library including in hot paths